app/src/main/java/com/bbuddies/madafaker/MadafakerFirebaseMessagingService.kt

```diff
@@ -0,0 +1,73 @@
+package com.bbuddies.madafaker
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Intent
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import com.bbuddies.madafaker.presentation.MainActivity
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+
+class MadafakerFirebaseMessagingService : FirebaseMessagingService() {
+
+    companion object {
+        private const val TAG = "MadafakerFirebaseMsgService"
+    }
+
+    override fun onMessageReceived(message: RemoteMessage) {
+        if (message.data.size > 0) {
+            Log.d(TAG, "Message Data payload: " + message.data)
+        }
+        if (message.notification != null) {
+            sendNotification(
+                message.notification!!.body, message.notification!!.title
+            )
+        }
+    }
+
+    private fun sendNotification(messageBody: String?, title: String?) {
```

Comment:
Nice, you have done basic notifications by yourself! Leave also a TODO: notification logic should be later encapsulated in NotificationManager. Need to create a task also.